src/transaction/log_manager.c

```diff
@@ -2748,22 +2747,9 @@ log_append_postpone (THREAD_ENTRY * thread_p, LOG_RCVINDEX rcvindex, LOG_DATA_AD
     {
       return;
     }
-  if (VACUUM_IS_THREAD_VACUUM (thread_p))
-    {
-      /* Cache postpone log record. Redo data must be saved before calling prior_lsa_next_record, which may free this
-       * prior node. */
-      vacuum_cache_log_postpone_redo_data (thread_p, node->data_header, node->rdata, node->rlength);
-      // todo - extend to all transactions
-    }
 
-  start_lsa = prior_lsa_next_record (thread_p, node, tdes);
-  if (VACUUM_IS_THREAD_VACUUM (thread_p))
-    {
-      /* Cache postpone log record. An entry for this postpone log record was already created and we also need to save
-       * its LSA. */
-      vacuum_cache_log_postpone_lsa (thread_p, &start_lsa);
-      // todo - extend to all transactions
-    }
+  // Cache postpone log record
+  tdes->m_log_postpone_cache.insert (*thread_p, *node, *tdes);
```

Comment:
I don't think current code is correct, it is possible that prior_lsa_next_record is never called. Better to call prior_lsa_next_record here instead of deferring to log_postpone_cache::insert.

```suggestion
  start_lsa = prior_lsa_next_record (thread_p, node, tdes);
  tdes->m_log_postpone_cache.insert (*thread_p, *node, start_lsa);
```

Reply:
prior_lsa_next_record might free redo_data from node, I'll go back to first implementation